routingv8/client.go

```diff
@@ -0,0 +1,149 @@
+package routingv8
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+const (
+	userAgent = "einride/here-go"
+)
+
+// RouteService handles communication with the route-related methods of the v7 HERE API.
+type RouteService service
+
+// MatrixService handles communication with the matrix-related methods of the v7 HERE API.
+type MatrixService service
+
+type Client struct {
+	// HTTP client used to communicate with the API.
+	client *http.Client
+
+	UserAgent string
+
+	// Reuse a single struct instead of allocating one for each service on the heap.
+	common service
+
+	// Route service.
+	Route *RouteService
+
+	// Matrix service.
+	Matrix *MatrixService
+}
+
+type service struct {
+	// URL for service API requests
+	URL    *url.URL
+	client *Client
+}
+
+// An ErrorResponse reports the error caused by an API request.
+type ErrorResponse struct {
+	// HTTP response that caused this error
+	Response *HereErrorResponse
+}
+
+func (r *ErrorResponse) Error() string {
```

Comment:
errors typically don't use upper case initial letters - any specific reason to do so here?

Reply:
No reason! Updated to lower case.